test/e2e/sched/sched_test.go

```diff
@@ -39,11 +42,38 @@ import (
 
 var _ = Describe("[Scheduler] imageReplacement", func() {
 	var initialized bool
+	nroSchedObj := &nropv1.NUMAResourcesScheduler{}
 	BeforeEach(func() {
 		if !initialized {
 			Expect(e2eclient.ClientsEnabled).To(BeTrue(), "failed to create runtime-controller client")
 		}
 		initialized = true
+		nroSchedKey := objects.NROSchedObjectKey()
+		Expect(e2eclient.Client.Get(context.TODO(), nroSchedKey, nroSchedObj)).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroSchedKey.String())
+
+		DeferCleanup(func() {
+			Eventually(func() bool {
+				err := e2eclient.Client.Get(context.TODO(), nroSchedKey, nroSchedObj)
+				if err != nil {
+					klog.Warningf("failed to get %q", nroSchedKey)
+					return false
+				}
+
+				nroSchedObj.Spec = objects.TestNROScheduler().Spec
```

Comment:
this works but it's fragile. We should save the Spec of the object we got and updated.
Not blocking though.